### DISCUSSION TITLE
Color of the rofi output is not preserved when piping it to the less.

### DISCUSSION THREAD

**Discussion by azizsayfiddin** [davatorium/rofi#1997] [Q&A] [answered]
When I run the `rofi -help | less -R`,  the color of the output is not preserved. How can I fix it?
OS: Arch/Hyprland
Rofi version: rofi-lbonn-wayland-git 1.7.5.wayland3.r26.gd88b475b-1

**Comment by DaveDavenport** [comment]
Rofi only outputs colors to a tty, this makes it easier to redirect the output to a file in a readable way.
There used to be  some tool that can 'fake' a tty when piping, but cannot remember the name.

**Reply by azizsayfiddin** [reply]
Have you remembered the name of that program?

**Reply by DaveDavenport** [reply]
could not remember it, but this works:

```bash
socat - EXEC:'rofi -h ',pty,setsid,ctty | less -R
```

**Reply by yusufaktepe** [reply, marked as answer]
> could not remember it

`unbuffer` maybe? (part of `expect`)

```sh
unbuffer rofi -h | less -R
```

**Reply by azizsayfiddin** [reply]
> > could not remember it
> 
> `unbuffer` maybe? (part of `expect`)
> 
> ```shell
> unbuffer rofi -h | less -R
> ```

Wow!

**Reply by azizsayfiddin** [reply]
> > could not remember it
> 
> `unbuffer` maybe? (part of `expect`)
> 
> ```shell
> unbuffer rofi -h | less -R
> ```

Where Did you find this program?

**Reply by yusufaktepe** [reply]
> Where Did you find this program?

I remembered I'd used it before, probably seen it at Stackoverflow.
